Authorization/Authorization/Presentation/SocialSign/SocialSignViewModel.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  SocialSignViewModel.swift
+//  Authorization
+//
+//  Created by Eugene Yatsenko on 11.10.2023.
+//
+
+import SwiftUI
+import Core
+import AuthenticationServices
+import FacebookLogin
+import GoogleSignIn
+import MSAL
+import Swinject
+
+enum SocialResult {
+    case apple(AppleCredentials)
+    case facebook(LoginManagerLoginResult)
+    case google(GIDSignInResult)
+    case microsoft(MSALAccount, String)
+
+    var backend: String {
+        switch self {
+        case .apple:
+            return "apple-id"
+        case .facebook:
+            return "facebook"
+        case .google:
+            return "google-oauth2"
+        case .microsoft:
+            return "azuread-oauth2"
+        }
+    }
+}
+
+final public class SocialSignViewModel: ObservableObject {
+
+    // MARK: - Properties -
+
+    private var onSigned: ((Result<SocialResult, Error>) -> Void)
+
+    init(
+        config: ConfigProtocol = Container.shared.resolve(ConfigProtocol.self) ?? ConfigMock(),
+        onSigned: @escaping (Result<SocialResult, Error>) -> Void
+    ) {
+        self.config = config
+        self.onSigned = onSigned
+    }
+
+    let  config: ConfigProtocol
+
+    private let appleSingInProvider: AppleSingInProvider  = .init()
+    private let googleSingInProvider: GoogleSingInProvider = .init()
+    private let facebookSingInProvider: FacebookSingInProvider = .init()
+    private let microsoftSingInProvider: MicrosoftSingInProvider = .init()
+
+    private var topViewController: UIViewController? {
+        UIApplication.topViewController()
+    }
+
+    // MARK: - Public Intens -
```

Comment:
can you change the `-` from the comments throughout? e.g. `// MARK: - Public Intens -` -> `// MARK: - Public Intens`

Reply:
why?